pylint/checkers/typecheck.py

```diff
@@ -1901,14 +1902,55 @@ def _detect_unsupported_alternative_union_syntax(self, node: nodes.BinOp) -> Non
             if not allowed_nested_syntax:
                 self._check_unsupported_alternative_union_syntax(node)
 
+    def _includes_version_compatible_overload(self, attrs: list[nodes.NodeNG]):
```

Comment:
```suggestion
    def _includes_version_compatible_overload(self, attrs: list) -> bool:
```
I'm not sure about the type for `attrs`. At the very least, it can also be `astroid.bases.BoundMethod` which doesn't inherit from `nodes.NodeNG`. Maybe lets leave it at just `list` for now.